src/UiPath.Workflow.Runtime/Hosting/WorkflowInstanceExtensionCollection.cs

```diff
@@ -118,7 +118,7 @@ internal WorkflowInstanceExtensionCollection(Activity workflowDefinition, Workfl
                 {
                     if (!TypeHelper.ContainsCompatibleType(allExtensionTypes, requiredType))
                     {
-                        throw FxTrace.Exception.AsError(new ValidationException(SR.RequiredExtensionTypeNotFound(requiredType.ToString())));
+                        throw FxTrace.Exception.AsError(new ExtensionRequiredException(requiredType));
```

Comment:
Introducing a breaking change for the most minimal need is not paying off at all.
We need a way to avoid breaking changes completely for this.

Reply:
One way around this would be to make the new exception an InnerException (also it may need to be serializable)